pkg/workloadapi/workload.proto

```diff
@@ -223,12 +223,35 @@ message GatewayAddress {
     NamespacedHostname hostname = 1;
     NetworkAddress address = 2;
   }
+
   // port to reach the gateway at for mTLS HBONE connections
   uint32 hbone_mtls_port = 3;
   // port to reach the gateway at for single tls HBONE connections
   // used for sending unauthenticated traffic originating outside the mesh to a waypoint-enabled destination
   // A value of 0 = unset
   uint32 hbone_single_tls_port = 4;
+
+  // (Optional) augment the inner stream upon decapsulation
+  ApplicationTunnel application_tunnel = 5;
```

Comment:
I'm pretty confident this must end up on GatewayAddress. We may have one Service that is plain HTTP and everything else on PROXY per this example from the capture doc: 

```yaml
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: waypoint
  namespace: hedgerow
spec:
  gatewayClassName: nginx
  listeners:
  - name: foo  
    # Will match listener based on rule #1 above
    hostname: foo.hedgerow.svc.cluster.local
    port: 80
    protocol: HTTP
    # All other traffic per both to-service and to-workload capture rules
  - name: catchall   
    port: 15008
    protocol: PROXY # or PROXY
```

If just had an ApplicationTunnel on the Waypoint's Workload, we'd have no way to tell which one of these we want (unless we want to just disallow a single Waypoint supporting both?)

Reply:
cc @howardjohn @louiscryan

Reply:
https://excalidraw.com/#json=_QuN06sEEbsEgQGtjTFrb,g30mQB51K3iKm-0mM889cw

Some of the other ideas..

Reply:
I think this is pretty good. Right now we've structured WDS to avoid needing extra lookups and this continues in that trend with a tidy addition for the new behavior. It also allows expression of something like, HBONE on a non-default port and then PROXY, which could be relevant.

Reply:
Offline chat:

```protobuf
// potentially embedded directly on the workload (for sandwich, this would be Gateway instances)
message InboundBindings {
   // option 1 for referencing; when we come inbound to any of these addrs, apply this config
   repeated NetworkAddress addresses;
   // option 2 for referencing
   repeated uid workloads;

  // keyed by hostname matcher (with wildcards) to pick the protocol and possible port override
  // similar to Gateway.Listeners from k8s
  // if not specified, inbound just copies bytes to the port specified in HBONE header
  // when specified, HBONE target must match one or traffic is denied
  map<string, ApplicationTunnel> bindings;
}
```